planner_cspace/src/planner_3d.cpp

```diff
@@ -441,76 +448,109 @@ class Planner3dNode
             continue;
           if (d.sqlen() > range_rough * range_rough)
             continue;
-          search_diffs.push_back(d);
+
+          SearchDiffs diffs;
+
+          const float grid_to_len = d.gridToLenFactor();
+          const int dist = d.len();
+          const float dpx = static_cast<float>(d[0]) / dist;
+          const float dpy = static_cast<float>(d[1]) / dist;
+          Astar::Vecf pos(0, 0, 0);
+          for (int i = 0; i < dist; i++)
+          {
+            Astar::Vec ipos(pos);
+            if (diffs.pos.size() == 0 || diffs.pos.back() != ipos)
+            {
+              diffs.pos.push_back(std::move(ipos));
+            }
+            pos[0] += dpx;
+            pos[1] += dpy;
+          }
+          diffs.grid_to_len = grid_to_len;
+          diffs.d = d;
+          search_diffs.push_back(std::move(diffs));
         }
       }
     }
 
-    while (true)
+    std::vector<Astar::PriorityVec> centers;
+    centers.reserve(num_cost_estim_task_);
+
+#pragma omp parallel
     {
-      if (open.size() < 1)
-        break;
+      std::vector<Astar::GridmapUpdate> updates;
+      updates.reserve(num_cost_estim_task_);
+
+      const float range_overshoot = ec_rough_[0] * (range_ + local_range_ + longcut_range_);
 
-      std::vector<Astar::PriorityVec> centers;
-      for (size_t i = 0; i < static_cast<size_t>(num_task_); ++i)
```

Comment:
Parallel task queue was not filled if L461 and L463 is not satisfied. It made parallelization overhead significant.